Documentation/etcd-mixin/mixin.libsonnet

```diff
@@ -21,12 +21,12 @@
                 sum without (%(etcd_instance_labels)s) (up{%(etcd_selector)s} == bool 0)
               or
                 count without (To) (
-                  sum without (%(etcd_instance_labels)s) (rate(etcd_network_peer_sent_failures_total{%(etcd_selector)s}[1m])) > 0.01
+                  sum without (%(etcd_instance_labels)s) (rate(etcd_network_peer_sent_failures_total{%(etcd_selector)s}[2m])) > 0.01
```

Comment:
I will admit to not understanding what is going on here vs. scrape rate, although the screenshots you'd posted [in the earlier PR][1] look good to me.  So I'm agnostic about this change.

I feel like I do understand the `for` increase you have below, and I approve of that change.

[1]: https://github.com/etcd-io/etcd/pull/12121#issuecomment-656832547

Reply:
Pretty sure `rate()` requires at least 2 samples to work (can't find a doc reference at the moment). @paulfantom claims in https://github.com/openshift/cluster-monitoring-operator/pull/853#discussion_r455709373 that 3 are required, but I'm not sure about. The general concern is that given a sample period of 30s, we're likely to have too few samples within a tight 1 minute window which I suspect will cause imprecision via extrapolation for the missing data points (please somebody correct me if this is incorrect).

Reply:
Makes sense for a lower bound.  What sets the upper bound?  Just "something less than half of `for`"?

Reply:
You need 2 samples for `rate()` to work but there is no guarantee that you will get two samples when "range = 2x scrape_interval". To make sure samples are in prometheus you need at least 3x and best practice says you need 4x to be resilient to potential scrape failure. Hence lower bound for range selector is 4x scrape_interval.

More on how to choose range in respect to scrape_interval in https://www.robustperception.io/what-range-should-i-use-with-rate

Reply:
So does this libsonnet need a variable where consumers can plug in their scrape rate?  Or do we just guess and say "surely nobody's scrape rate is bigger than $whatever"?

Reply:
Also, range upper bound from the doc you link:

> If you take a rate over an hour and alert on that, that's all well and good until the underlying condition stops and then you have to wait an hour until the alert goes away.

So I think this should be "4x scrape rate", and we can extend `for` (which doesn't have the recovery-latency issue) as necessary to minimize false positives (while trying to avoid _crazy_ amounts of discovery latency).

Reply:
With 9aa5d47 I tried implementing a new scrape interval configuration and changed the rate window to be a function of the interval ("4x scrape interval") — how does it look? I couldn't find an easier way to do the arithmetic inside the string interpolation itself, so any refactoring tips welcome.

Reply:
@wking @paulfantom does this seem better? I'm not sure what the practices are around introducing new config.

Reply:
lgtm :+1:

Reply:
@wking @paulfantom while revising the unit tests, it occurred to me that the `for: 10m` change also means that, because of the `or` condition in the expression, we're also going to endure 10 minutes of `up=0` before firing even though that's a much less ambiguous condition.

Think this is a necessary compromise given everything else?

Reply:
Unit tests all fixed up